compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIEDistributeCoresAndObjectFifos.cpp

```diff
@@ -136,24 +136,58 @@ LogicalResult distributeLocalMemory(ModuleOp moduleOp) {
     //                     AllocOp and not replace the current AllocOp's use
     //                     within.
     memref::AllocOp newAlloc = memrefToNew[allocOp];
+    ArrayRef<int64_t> newAllocShape = newAlloc.getType().getShape();
     for (Operation *userOp : allocOpUsers) {
       if (auto subviewOp = dyn_cast<memref::SubViewOp>(userOp)) {
         rewriter.replaceAllUsesWith(subviewOp, newAlloc);
         toBeErased.push_back(subviewOp);
       } else if (auto transferReadOp =
                      dyn_cast<vector::TransferReadOp>(userOp)) {
         rewriter.setInsertionPoint(transferReadOp);
-        Value newTransferReadOp = rewriter.create<vector::TransferReadOp>(
+        // For dimensions with size as 1 we set the indices as 0.
+        SmallVector<Value> newIndices = transferReadOp.getIndices();
+        Value c0 =
+            rewriter.create<arith::ConstantIndexOp>(transferReadOp.getLoc(), 0);
+        for (unsigned i = 0, n = newAllocShape.size(); i < n; i++) {
+          if (newAllocShape[i] == 1) newIndices[i] = c0;
+        }
+        auto newTransferReadOp = rewriter.create<vector::TransferReadOp>(
             transferReadOp.getLoc(), transferReadOp.getType(), newAlloc,
-            transferReadOp.getIndices());
-        rewriter.replaceAllUsesWith(transferReadOp, newTransferReadOp);
+            newIndices);
+        // The old vector.transfer_read has all `in_bounds` set as `true`. We
+        // therefore need to do the same on this new vector.transfer_read op
+        // else this would yield a masked load later on in the lowering stack.
+        SmallVector<bool, 4> bools(newTransferReadOp.getTransferRank(), true);
+        auto inBoundsAttr = rewriter.getBoolArrayAttr(bools);
```

Comment:
Does TransferReadOp have a build method which takes attributes? Something like 

```
  auto newTransferReadOp = rewriter.create<vector::TransferReadOp>(
            transferReadOp.getLoc(), transferReadOp.getType(), newAlloc, newIndices, tranferReadOp.getAttributes());
```

might work

Reply:
I agree, it should get the attribute from the old transferReadOp.